### ISSUE TITLE
New Rule: no-invalid-charset-placement

### ISSUE THREAD

**Issue by thecalamiity** [eslint/css#163]
### Rule details

Disallow invalid placement of @charset rule.

### What type of rule is this?

Warns about a potential problem

### Example code

```css
a {}
@charset "utf-8";
```

### Prior Art

```markdown
I couldn't find a Stylelint equivalent, but VS Code's built-in linter reports this.
```

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

`@charset` can only be placed at the very beginning of a stylesheet. No other characters, except the Unicode byte-order mark, are allowed before it.
